src/ui/public/agg_types/controls/top.html

```diff
@@ -0,0 +1,29 @@
+<div class="form-group">
+  <label for="sort">
+    Sort On
+  </label>
+
+  <select
+    class="form-control"
+    name="sortField"
+    required
+    ng-model="agg.params.sortField"
+    ng-show="indexedFields.length"
+    auto-select-if-only-one="indexedFields"
+    ng-options="field as field.displayName group by field.type for field in indexedFields">
+  </select>
```

Comment:
Ah, I see now that this really maps directly to the sort param on the top_hits agg. I think it makes sense to keep them separate then, this code is will surely end up having different concerns than the generic field directive.